api/python/cellxgene_census/src/cellxgene_census/experimental/pp/_highly_variable_genes.py

```diff
@@ -88,14 +88,13 @@ def _highly_variable_genes_seurat_v3(
         mvn = MeanVarianceAccumulator(n_batches, n_samples, query.n_vars)
         for arrow_tbl in _EagerIterator(query.X(layer).tables(), pool=pool):
             data = arrow_tbl["soma_data"].to_numpy()
+            var_dim = var_indexer.by_var(arrow_tbl["soma_dim_1"])
             if batch_indexer:
                 _batch_take_at_future = pool.submit(batch_indexer, arrow_tbl["soma_dim_0"])
-                var_dim = var_indexer.by_var(arrow_tbl["soma_dim_1"])
                 _batch_vec = batch_codes[_batch_take_at_future.result()]
-                mvn.update_by_batch(_batch_vec, var_dim, data)
             else:
-                var_dim = var_indexer.by_var(arrow_tbl["soma_dim_1"])
-                mvn.update_single_batch(var_dim, data)
+                _batch_vec = None
+            mvn.update(var_dim, data, _batch_vec)
```

Comment:
I simplified the code a bit here, so that the `update` method is only called once.

Reply:
this simplification removes the parallelism (by moving the var reindexing up, it completely serializes everything - the thread submit of the row indexer is wasted overhead).

Either one of these is fine (although the second has a performance hit):
1. move the var reindexing so it remains in-parallel with row indexing
2. or, get rid of the thread submit for row indexing

Reply:
Ah, that's interesting to know. I will fix it. Thanks!